app/misc/git.ts

```diff
@@ -1534,39 +1534,38 @@ function cleanRepo() {
 }
 
 /**
- * This method is called when the sync button is pressed, and causes the fetch-modal
- * to appear on the screen.
+ * This method is called when the user clicks the "Remote" button on the navbar.
  */
-function requestLinkModal() {
-  $("#fetch-modal").modal();
+function setUpstreamModal() {
+  $('#set-upstream-modal').modal('show');
 }
 
 /**
- * This method is called when a valid URL is given via the fetch-modal, and runs the
- * series of git commands which fetch and merge from an upstream repository.
+ * Clears the fields from the upstream repo modal.
  */
-function fetchFromOrigin() {
-  console.log("begin fetching");
-  let upstreamRepoPath = document.getElementById("origin-path").value;
-  if (upstreamRepoPath != null) {
+function clearUpstreamModalText() {
+  document.getElementById("remote-path").value = "";
+}
+
+/**
+ This function is called after the user enters the address of the upstream repo to sync from. It sets the
+ entered address as the upstream repo of the forked repository.
+ */
+function setUpstreamRepo() {
+  let repository;
+  let upstreamRepoPath = document.getElementById("remote-path").value;
+  if(upstreamRepoPath != null) {
     Git.Repository.open(repoFullPath)
       .then(function (repo) {
-        console.log("fetch path valid")
-        displayModal("Beginning Synchronisation...");
-        addCommand("git remote add upstream " + upstreamRepoPath);
-        addCommand("git fetch upstream");
-        addCommand("git merge upstrean/master");
-        console.log("fetch successful")
-        updateModalText("Synchronisation Successful");
-        refreshAll(repo);
-      },
-        function (err) {
-          console.log("Waiting for repo to be initialised");
-          displayModal("Please select a valid repository");
-        });
-  } else {
-    displayModal("No Path Found.")
+      repository = repo;
+      var result = Git.Remote.createWithFetchspec(repository, 'upstream', upstreamRepoPath, '+refs/heads/*:refs/remotes/upstream/*');
+      addCommand("git remote add upstream " + upstreamRepoPath);
+      console.log(result)
+    }, function(err) {
```

Comment:
If I add throw new Error('test error'); to line 1560 I do not see any error :(. Please fix the error handling to show a dialog if an error occurs.